File: native-with-db-server/app/src/main/java/ubb/barcoaie/lab5_ma/activities/Create/CreateActivity.java
package ubb.barcoaie.lab5_ma.activities.Create;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import ubb.barcoaie.lab5_ma.model.Vegetable;
import ubb.barcoaie.lab5_ma.R;
import ubb.barcoaie.lab5_ma.repository.VegetableDatabase;


public class CreateActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_create);

        EditText vegetable_name = this.findViewById(R.id.vegetable_name);
        EditText vegetable_price = this.findViewById(R.id.vegetable_price);

        Button create_button = this.findViewById(R.id.button);
        create_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent reply = new Intent();
                Vegetable vegetable = new Vegetable(vegetable_name.getText().toString(), Integer.parseInt(vegetable_price.getText().toString()), -1);
                VegetableDatabase.databaseWriteExecutor.execute(() -> {
                    VegetableDatabase.getDatabase(v.getContext()).vegetableDao().insert(vegetable);
                });
                reply.putExtra("add", vegetable);
                setResult(RESULT_OK, reply);
                CreateActivity.this.finish();
            }
        });
    }

    @Override
    public void onBackPressed() {
        Intent replyIntent = new Intent();
        setResult(RESULT_CANCELED, replyIntent);
        CreateActivity.this.finish();
    }
}

File: native/app/src/main/java/ubb/barcoaie/lab2_ma/Activities/CreateActivity.kt
package ubb.barcoaie.lab2_ma.Activities

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import com.google.android.material.textfield.TextInputEditText
import ubb.barcoaie.lab2_ma.Domain.Vegetable
import ubb.barcoaie.lab2_ma.R
import ubb.barcoaie.lab2_ma.Repo.VegetableRepoImpl

class CreateActivity : AppCompatActivity() {
    var repo = VegetableRepoImpl()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_create)

        val addButton: Button = findViewById(R.id.add_button)

        val input_price: TextInputEditText = findViewById(R.id.price2)
        val input_name: TextInputEditText = findViewById(R.id.name)

        repo = intent.extras?.get("Repo") as VegetableRepoImpl

        addButton.setOnClickListener {
            val input_name_text = input_name.text.toString()
            val input_price_text = input_price.text.toString().toInt()
            repo.addVegetable(Vegetable(input_name_text, input_price_text))

            println("coae merge")
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        val intent = Intent(this, MainActivity::class.java)
        intent.putExtra("Repo", repo)
        startActivity(intent)
    }
}